### DISCUSSION TITLE
Gioteck's Dualshock3 clone does not work

### DISCUSSION THREAD

**Discussion by cualquiercosa327** [cathery/sys-con#93] [Controller requests]
Hello,first excuse me about spamming on other threads: it was no my intention.
I have modded your code to try make my pad works (it is a private mod because i prefer use this gamepad to the other dualshock 3 clone i own : i dont care make the original doesnt work on my switch.I also have create other version of the code for avoid broken normal ds3 compatibility which is on my github but it is still no finished because i have two error while compiling)
here is the model:

https://github.com/cathery/sys-con/issues/80

i have changed the Vid and Pid with Gioteck one and the switch see it as a gamepad on the menu but it doesnt work fine.
i test also  on your applet and i have found than the problem is than FSDevice is changing constantly,increasing with 1 ,i mean : when i open again the applet it is the latest FSDevice number increased+ 1.

Excuse me : i upload here my sources because i can no do on my github where i have a third version which still is no finished.

Other issue with my code is : 

if i conect it as first gamecontroller and joycon as the second ,it has some issues .
If the joycon is the first,i could use the switch but the controller doesnt work fine (sometimes it register some button ,others no)

The dualshock4 works without problem on both versions of my code.

[modded.zip](https://github.com/cathery/sys-con/files/3882309/modded.zip)

I decided post it here because i thought maybe it could have some relation with the issue/bug 
of https://github.com/cathery/sys-con/issues/78.

Ah,a last request : could you post your applet source code for studing it?

Thank you and excuse me again.

**Comment by cathery** [comment]
Latest commits might fix the issues you had with the dualshock 3

**Comment by cualquiercosa327** [comment]
Hello,thanks.I have tested it and it has the same problem .
Also it seems make the module more inestable , explain : when i connect it with the new version (compiled for me with my little file changes) ,i try use a joycon to enter your app for testing and it doesnt work.
At last the way i could make test was open the app and later conect it on the usb.
Curiously i could use the joycon to press y ,exit the app and enter again.
I must make more testing for assure it is more inestable than the older version ( i tried 3 times,rebooting the switch) 

Curiously when  i conect the ds3 on linux,the Gioteck one is detected and create the devices needed for use it.I could test it with jstest,evtest..

But the other,the one which works flawlessly with your module,give some errors and doesnt work. 

(i can attach the output from both if you want)

While i searched some info,I found than they are lot of people who bought what they thought was a original dualshock 3 for later found than it was a clone. An interesting example  is on:

https://forums.gentoo.org/viewtopic-t-1038906-start-0.html

the amazing of this is than most of them is they share the same  Manufacturer : ShanWan (The same as my two clones,but i knew they were clone and i paid less).

I am looking on the linux kernel module source code if i found why the  Gioteck doesnt work with sys-con.

if my memory doesnt fail,you post than you had a ds3 clone which also doesnt work with sys-con.
could you look if it is also a "Shanwan" ?
